include/fiction/networks/views/extended_rank_view.hpp

```diff
@@ -398,6 +410,33 @@ class extended_rank_view<Ntk, false> : public fiction::depth_view<Ntk>
                   [this](auto const& n1, auto const& n2) { return rank_pos.at(n1) < rank_pos.at(n2); });
         mockturtle::detail::foreach_element(pis.cbegin(), pis.cend(), std::forward<Fn>(fn));
     }
+
+    /**
+     * Applies a given function to each PI in unranked order, so the order in the underlying `depth_view`.
+     *
+     * @tparam Fn Functor type.
+     * @param fn The function to apply.
+     */
+    template <typename Fn>
+    void foreach_pi_unranked(Fn&& fn) const
+    {
+        fiction::depth_view<Ntk>::foreach_pi(std::forward<Fn>(fn));
+    }
```

Comment:
The docstring isn't quite clear to me. It seems to miss a few words grammatically. Do you mean something like

"Applies a given function to each PI in unranked order. That means the order of the underlying `depth_view`."

?

If that is the case, what would that "order of the underlying `depth_view`" be? The order of PI creation? Could you add that?